feat: add self service refund of fare contracts

rosvik opened the pull request:
https://github.com/user-attachments/assets/761c3f76-fa63-41cd-a0f3-5903c6c015a0

closes https://github.com/AtB-AS/kundevendt/issues/20058

marius-at-atb commented on src/modules/fare-contracts/components/RefundBottomSheet.tsx:
```diff
@@ -0,0 +1,111 @@
+            type="error"
+          />
+        )}
+        <Section>
+          <GenericSectionItem type="spacious">
+            <ThemeText>
+              {t(FareContractTexts.refund.bottomSheetDescription)}
+            </ThemeText>
+          </GenericSectionItem>
+        </Section>
```
UX question - should this be hidden when `refundOptions?.is_refundable === false`?

rosvik replied:
Had a talk with @hannahATB, and no, we want to show the info even when the ticket is not available for refund.

marius-at-atb commented on src/modules/fare-contracts/components/RefundBottomSheet.tsx:
```diff
@@ -0,0 +1,111 @@
+        />
+        <Button
+          backgroundColor={theme.color.background.neutral[1]}
+          onPress={() => {
+            close();
+          }}
```
Can be just `onPress={close}`

rosvik replied:
```suggestion
          onPress={close}
```

marius-at-atb commented on src/ticketing/types.ts:
```diff
@@ -193,3 +193,11 @@ export type AddPaymentMethodResponse = {
   terminal_url: string;
 };
+
+/**
+ * Defined by RefundOptionsResponse in
+ * https://github.com/AtB-AS/sales/blob/main/sales-service/src/handlers/sales/refund.rs#L31
+ */
```
Good to have comments like this :+1:

rosvik replied:
I guess the line number will change over time though, so removing that for now ✌️

```suggestion
/**
 * Defined by RefundOptionsResponse in
 * https://github.com/AtB-AS/sales/blob/main/sales-service/src/handlers/sales/refund.rs
 */
```

marius-at-atb reviewed (approved):
Very nice!
I guess you can remove "wip" from the PR title now? 😄

rosvik commented:
Slight change of plans here, and we decided to only show the refund option if the ticket is refundable. Updated screen recording. the major change since last review is here: f05c30f26eaa046b2ec851e38fd40b3cef2e9070

reidzeibel reviewed (approved):
new changes looks good!
